### PR TITLE
fixed mongo issues related to 3.6.x

### LINKED ISSUE #793: MongoDB collection displays as $cmd when using a cursor on collection.aggregate
[NOTE]: # ( ^^ Provide a general summary of the issue in the title above. ^^ )

## Description

[NOTE]: # ( Describe the problem you're encountering. )
[TIP]:  # ( Do NOT share sensitive information, whether personal, proprietary, or otherwise! )

The mongodb versioned tests will fail if you run `collection-misc.tap.js` on versioned > 3.3.0.  
[Failure Thread](https://github.com/newrelic/node-newrelic/pull/792#discussion_r664574060)

It appears the mongodb package was refactored and any cursor operation done on `collection.aggregate` will display the collection as `$cmd` as it's getting set as the `this.ns` in [command_v2](https://github.com/mongodb/node-mongodb-native/blob/3.6/lib/operations/command_v2.js#L20)

## Expected Behavior

[NOTE]: # ( Tell us what you expected to happen. )

The appropriate collection should be displayed for the cursor segment but instead shows as `$cmd`

## [Troubleshooting](https://discuss.newrelic.com/t/troubleshooting-frameworks/108787) or [NR Diag](https://docs.newrelic.com/docs/using-new-relic/cross-product-functions/troubleshooting/new-relic-diagnostics) results

[NOTE]: # ( Provide any other relevant log data. )
[TIP]:  # ( Scrub logs and diagnostic information for sensitive information )

## Steps to Reproduce
Run the `collection-misc.tap.js` tests in `test/versioned/mongodb` on mongodb version 3.6.9(as an example)

[NOTE]: # ( Please be as specific as possible. )
[TIP]:  # ( Link a sample application that demonstrates the issue. )

## Additional context

[TIP]:  # ( Add any other context about the problem here. For example, relevant community posts or support tickets. )
If this code is added to the `setParser` method the one failure test will pass and all others are unaffected

```
--- a/lib/instrumentation/mongodb.js
+++ b/lib/instrumentation/mongodb.js
@@ -112,6 +112,8 @@ function initialize(agent, mongodb, moduleName, shim) {
       collection = this.collection.collectionName
     } else if (this.s && this.s.name) {
       collection = this.s.name
+    } else if (this.operation && this.operation.target) {
+      collection = this.operation.target
     } else if (this.ns) {
       collection = this.ns.split(/\./)[1] || collection
     }
```

### LINKED ISSUE #794: Triage db.tap.js mongodb versioned tests > 3.2.6
## Is your feature request related to a problem? Please describe.
The mongodb db.tap.js tests are failing on versions > 3.2.6

## Feature Description
It looks like behavior in mongodb has changed around addUser and createCollection.  addUser no longer returns the newly created user in the callback, just `{ ok: 1 }`.  createCollection now returns an error if the collection already exists.  Both of these behaviors break our test suite which will require some refactoring of the tests.  I don't think this will affect the instrumentation.

## Additional context
To see this failure just install mongodb@3.6.9 and run `db.tap.js`, it initially hangs because of an uncaught exception around `t.equal(user[0].user, userName)`.  If you remove this assertion the `createCollection` tests will fail because the 2nd test fails around the collection already being present

## Priority
Please help us better understand this feature request by choosing a priority from the following options: 
[Really Want]

### PR DESCRIPTION
**Pull request by bizob2828**
## Proposed Release Notes
 * Added additional instrumentation to properly set the collection on versions `>= 3.3.0`
 * Updated mongo versioned stanzas to run all tests on versions `<4`

## Links
Closes #794 
Closes #793

### CODE REVIEW

File: test/versioned/mongodb/db.tap.js [hunk 1]
```diff
@@ -63,20 +63,18 @@ if (semver.satisfies(mongoPackage.version, '<3')) {
...
-  var userPass = 'user-test-pass'
+  const userName = 'user-test'
+  const userPass = 'user-test-pass'
 
   db.removeUser(userName, function preRemove() {
     // Don't care if this first remove fails, it's just to ensure a clean slate.
     db.addUser(userName, userPass, {roles: ['readWrite']}, added)
   })
 
-  function added(err, _user) {
```
Comment (carlo-808) [inline test/versioned/mongodb/db.tap.js:75]: Did the call back signature change?

**Review by carlo-808 — COMMENTED** [review]
All seems good, just added a couple comments.

**Review by carlo-808 — APPROVED** [review]

File: test/versioned/mongodb/db.tap.js (hunk 1 above)
Comment (bizob2828) [reply test/versioned/mongodb/db.tap.js:75]: yea so it no longer returns data, just `{ ok: 1 }`. I figured the assertions later down the chain were pointless anyways

File: test/versioned/mongodb/db.tap.js [hunk 2]
```diff
@@ -328,11 +329,16 @@ function dbTest(name, collections, run) {
     // to connect to, which only happens if there is a Mongo instance running on
     // the same box as these tests. This should always be the case on Travis,
     // but just to be sure they're running there check for the environment flag.
-    var shouldTestDomain = domainPath || process.env.TRAVIS
+    const shouldTestDomain = domainPath || process.env.TRAVIS
```
[the remark this answers is not in the document]
Comment (bizob2828) [reply test/versioned/mongodb/db.tap.js:332]: ah, good catch. ill do that

**Review by michaelgoin — APPROVED** [review]
👍

**Pull request merged**